### PR TITLE
Code style fixes in tests

### PR DESCRIPTION
**Pull request by glensc**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Applies code style fixes to tests:
- add return types
- add parameter types
- cleanup dead code
- fixup phpdoc

... see commits for details

### CODE REVIEW

**Comment by glensc** [comment]
How do you fix such CS failure?

```
 184 | WARNING | Line exceeds 120 characters; contains 123 characters
```

```
     * @param  array|object|Config|Protocol\Imap $params mail reader specific parameters or configured Imap protocol object
```

- https://travis-ci.com/github/laminas/laminas-mail/jobs/375625490

**Comment by glensc** [comment]
if needed, can move some changes to separate PRs!

**Comment by glensc** [comment]
@weierophinney ping!

**Review by Ocramius — APPROVED** [review]
This totally went under the radar: LGTM as-is, thanks @glensc!

**Comment by Ocramius** [comment]
Hmm, yeah, still needs a rebase, but the remaining CS issues could be ignored/suppressed IMO, since this is already a massive improvement

**Comment by glensc** [comment]
rebased, further changes can be in separate PR

**Comment by Ocramius** [comment]
Will try getting this merged later today 👍

**Comment by glensc** [comment]
The cs errors about line length remain:
- https://travis-ci.com/github/laminas/laminas-mail/jobs/466475515

```
FILE: /home/travis/build/laminas/laminas-mail/src/Storage/Pop3.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 125 | WARNING | Line exceeds 120 characters; contains 123 characters
----------------------------------------------------------------------
FILE: /home/travis/build/laminas/laminas-mail/src/Storage/Imap.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 184 | WARNING | Line exceeds 120 characters; contains 123 characters
----------------------------------------------------------------------
FILE: /home/travis/build/laminas/laminas-mail/test/MessageTest.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 762 | WARNING | Line exceeds 120 characters; contains 121 characters
----------------------------------------------------------------------
```

how to suppress these? the error unfortunately isn't helpful in that sense.

I would personally change line length globally to 200, but that probably needs to be done then in `vendor/laminas/laminas-coding-standard/ruleset.xml`.

**Comment by Ocramius** [comment]
> `Line exceeds 120 characters; contains 123 characters`

Honestly, these are minor, and inline suppression with phpcs via (I think) `// @phpcs:disable` and then `// @phpcs:enable` should work

**Comment by glensc** [comment]
Added phpcs:disable / phpcs:enable, oh and sign-off thing is annoying

**Comment by glensc** [comment]
Also, coveralls is wrong, this is not decrease of coverage. it's the same code:

```diff
--- a/src/Storage/Writable/Maildir.php
+++ b/src/Storage/Writable/Maildir.php
@@ -51,9 +51,9 @@ class Maildir extends Folder\Maildir implements WritableInterface
                     throw new StorageException\InvalidArgumentException("parent $dir not found", 0, $error);
                 } elseif (! is_dir($dir)) {
                     throw new StorageException\InvalidArgumentException("parent $dir not a directory", 0, $error);
-                } else {
-                    throw new StorageException\RuntimeException('cannot create maildir', 0, $error);
                 }
+
+                throw new StorageException\RuntimeException('cannot create maildir', 0, $error);
             }
         }

@@ -390,10 +390,12 @@
```

**Comment by glensc** [comment]
The phpcs:disable has no effect, but it works in here:
- https://github.com/laminas/laminas-validator/pull/62/files

**Comment by glensc** [comment]
`phpcs:ignore Generic.Files.LineLength.TooLong` doesn't work either. I don't know how to place it. nothing makes sense.

the only thing that worked was to disable the whole check for file:
- https://github.com/laminas/laminas-validator/commit/a5d54fa438f028d05f49d365abac2e91604043eb

so, I'm removing the changes that break the limit, to be able to merge this PR:
- f0eea2a98cd359b37f9c5345d074d91c96272cbb

**Comment by glensc** [comment]
@Ocramius can you have another look? Travis green now, Coveralls still red, because it's stupid as outlined above.

**Comment by Ocramius** [comment]
No worries about coveralls: more a signal than a binding flag.

File: src/Protocol/Pop3.php [hunk 1]
```diff
@@ -127,7 +127,7 @@ public function connect($host, $port = null, $ssl = false)
     public function sendRequest($request)
     {
         ErrorHandler::start();
-        $result = fputs($this->socket, $request . "\r\n");
+        $result = fwrite($this->socket, $request . "\r\n");
```
Comment (Ocramius) [inline src/Protocol/Pop3.php:130]: Possibly risky change: re thr semantics the same?

**Review by Ocramius — APPROVED** [review]
LGTM 👍 looking up `fputs()` vs `fwrite()` semantics before merging

File: src/Protocol/Pop3.php (hunk 1 above)
Comment (Ocramius) [reply src/Protocol/Pop3.php:130]: Ok, they are aliases: 🚢

**Comment by Ocramius** [comment]
Thanks @glensc!

**Comment by glensc** [comment]
fputs is alias to fwrite: https://www.php.net/fputs

**Comment by glensc** [comment]
Extracted the one change to separate PR:
- https://github.com/laminas/laminas-mail/pull/126

so, somebody could try to handle the setup of exception for line length.

**Pull request merged**